modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/MuleExtensionAnnotationParser.java

```diff
@@ -0,0 +1,227 @@
+            {
+                parseGroupParameters(parameterTypes[i], parameterDescriptors);
+            }
+            else
+            {
+                ParameterDescriptor parameterDescriptor = doParseParameter(paramNames[i], parameterTypes[i], annotations);
+                if (parameterDescriptor != null)
+                {
+                    parameterDescriptors.add(parameterDescriptor);
+                }
+            }
+        }
+
+        return parameterDescriptors;
+    }
+
+    private static void parseGroupParameters(DataType parameterType, List<ParameterDescriptor> parameterDescriptors)
+    {
+        for (Field field : IntrospectionUtils.getParameterFields(parameterType.getRawType()))
+        {
+            if (field.getAnnotation(org.mule.extension.annotations.ParameterGroup.class) != null)
+            {
+                parseGroupParameters(getFieldDataType(field), parameterDescriptors);
+            }
+            else
+            {
+                ParameterDescriptor parameterDescriptor = doParseParameter(field.getName(), getFieldDataType(field), toMap(field.getAnnotations()));
+                if (parameterDescriptor != null)
+                {
+                    parameterDescriptors.add(parameterDescriptor);
+                }
+            }
+        }
+    }
+
+    private static ParameterDescriptor doParseParameter(String paramName, DataType parameterType, Map<Class<? extends Annotation>, Annotation> annotations)
+    {
+        if (IMPLICIT_ARGUMENT_TYPES.contains(parameterType.getRawType()))
+        {
+            return null;
+        }
+
+        DataType dataType = parameterType;
+
+        ParameterDescriptor parameter = new ParameterDescriptor();
+        parameter.setName(getParameterName(paramName, (Parameter) annotations.get(Parameter.class)));
+        parameter.setType(dataType);
+
+        Optional optional = (Optional) annotations.get(Optional.class);
+        if (optional != null)
+        {
+            parameter.setRequired(false);
+            parameter.setDefaultValue(getDefaultValue(optional));
+        }
+        else
+        {
+            parameter.setRequired(true);
+        }
+
+        Payload payload = (Payload) annotations.get(Payload.class);
+        if (payload != null)
+        {
+            parameter.setRequired(false);
+            parameter.setDefaultValue("#[payload]");
+            parameter.setHidden(true);
+        }
+
+        RestrictedTo typeRestriction = (RestrictedTo) annotations.get(RestrictedTo.class);
+        if (typeRestriction != null)
+        {
+            parameter.setTypeRestriction(typeRestriction.value());
+        }
+        return parameter;
+    }
+
+    public static String[] getParamNames(Method method)
+    {
+        String[] paramNames;
+        try
+        {
+            paramNames = new ParamReader(method.getDeclaringClass()).getParameterNames(method);
```

Comment:
Yes it is. Also as a side note. once we start requiring JDK 8 we won't need this anymore